internal/consensus/state.go

```diff
@@ -1620,30 +1620,69 @@ func (cs *State) defaultDoPrevote(ctx context.Context, height int64, round int32
 		return
 	}
 
-	// If we're not the proposer, we need to build the block
-	if cs.config.GossipTransactionKeyOnly && cs.Proposal != nil && cs.ProposalBlock == nil {
-		txKeys := cs.Proposal.TxKeys
-		if cs.ProposalBlockParts.IsComplete() {
+	if cs.config.GossipTransactionKeyOnly {
+		if cs.ProposalBlock == nil {
+			// If we're not the proposer, we need to build the block
+			txKeys := cs.Proposal.TxKeys
+			if cs.ProposalBlockParts.IsComplete() {
+				block, err := cs.getBlockFromBlockParts()
+				if err != nil {
+					cs.logger.Error("Encountered error building block from parts", "block parts", cs.ProposalBlockParts)
+					return
+				}
+				// We have full proposal block and txs. Build proposal block with txKeys
+				proposalBlock := cs.buildProposalBlock(height, block.Header, block.LastCommit, block.Evidence, block.ProposerAddress, txKeys)
+				if proposalBlock == nil {
+					return
+				}
+				cs.ProposalBlock = proposalBlock
+			} else {
+				return
+			}
+		}
+	} else {
+		if cs.ProposalBlock == nil {
 			block, err := cs.getBlockFromBlockParts()
 			if err != nil {
 				cs.logger.Error("Encountered error building block from parts", "block parts", cs.ProposalBlockParts)
 				return
 			}
-			// We have full proposal block and txs. Build proposal block with txKeys
-			proposalBlock := cs.buildProposalBlock(height, block.Header, block.LastCommit, block.Evidence, block.ProposerAddress, txKeys)
-			if proposalBlock == nil {
+			if block == nil {
+				logger.Debug("prevote step: ProposalBlock is nil")
+				cs.signAddVote(ctx, tmproto.PrevoteType, nil, types.PartSetHeader{})
```

Comment:
if we add this here then we probably don't need this part anymore right? 
https://github.com/sei-protocol/sei-tendermint/pull/26/files#diff-c27a58d5a8b3c15166169543b3cdb94da91bb60efa27609ef446719474af78c5R1682-R1687

Reply:
yeah that's right. Removed